layouts: apply design system

BibianaC opened the pull request:
**Latest: 15/01/2026**

<img width="1905" height="1628" alt="localhost_3000_" src="https://github.com/user-attachments/assets/f3a4e978-ffc4-4c12-813b-7f9aaa9cbf9a" />

BibianaC committed:
layouts: apply design system - draft

BibianaC committed:
layouts: apply design system: footer

BibianaC commented:
@markbrough I have started implementing the design system. I'm stuck with the about and language dropdowns. 
I have a question about how much should be translated in the footer and the header.

stevieflow commented:
@BibianaC 

- Would the About drop-down become the standard Help Docs button we have in the design system, pointing to https://docs.countrydata.iatistandard.org/en/latest/ (cc/ @isabelbirds) ?
- For languages, can we use the existing languages that CDFD provides?
- For the footer, we have live examples of how these look in Spanish and French via the Validator docs: https://docs.validator.iatistandard.org/fr/latest/ - can they be used? 
- (we don't have a Portugese footer though --> @robredpath)

Thanks

IsabelBirds commented:
Yes please - couple of updates to make but the docs should be live and available via the standard set up.

codemacabre committed:
layouts: add changeLocale method

codemacabre commented:
In doing a little work on this I noticed that there currently isn't a navigation menu dropdown in the design system (a different type of dropdown to the `select` element I recently added). Rather than create a custom one for this project, should I create one in the design system first, then use it here once it's been released? Won't take long at all.

cc. @robredpath @stevieflow  @BibianaC

robredpath commented:
@codemacabre can you say a bit more about what that is? Do you mean as in when you click on an item in primary navigation, a dropdown appears with further (nested) places you could go? 

We didn't create a nested menu design when we did the first round of work on the design system because it wasn't needed for the initial products and we wanted to give the whole topic some proper consideration as it would become the standard for all our products so we wanted to make a good choice. 

Most of our products actually don't need deeply nested menus, and even the Dashboard (which used to have a sprawling menu structure!) has I think been improved by replacing menus with boxes (for now) and fancy boxes (when we get around to implementing them)

Where does CDFD use this? Can we change CDFD to be more like this emerging trend across the rest of the web estate? We we need it we need it, but I figured it's good to ask this question first!" 

@stevieflow when we're back from the MA I'll get a PT translation of the relevant bits arranged,

codemacabre commented:
@robredpath This is to replace the design of the existing dropdown 'About' menu:

<img width="306" height="201" alt="image" src="https://github.com/user-attachments/assets/66d92301-5509-464e-9196-36b92b8abe5b" />

If you'd like to reorganise the menu design instead, please let me know what you'd prefer it to be. The current (incomplete) design is as follows:

<img width="618" height="274" alt="image" src="https://github.com/user-attachments/assets/c02eb76b-d242-40cf-98ee-6a1832b6fb2f" />

stevieflow commented:
All the items from the About drop down should now be in the Help Docs @IsabelBirds ?

IsabelBirds commented:
Yes the about menu is not needed - just the standard link to the help docs please.

codemacabre commented:
Great stuff, that saves doing a bit of extra work. Thanks all.

codemacabre committed:
config: update .gitignore, upgrade to latest design system

codemacabre committed:
config: update dependencies

codemacabre committed:
layouts: change about menu to single link

codemacabre committed:
content: index: update primary buttons to use iati-button classes

stevieflow commented:
@robredpath do you think we can get this checked / merged before the year is out?

BibianaC commented:
@robredpath, this is how it looks:

<img width="758" height="64" alt="Screenshot 2026-01-13 at 11 26 50" src="https://github.com/user-attachments/assets/8365cc14-4ebd-4e39-aada-fa43d7581ec9" />
<img width="1669" height="959" alt="Screenshot 2026-01-13 at 11 27 02" src="https://github.com/user-attachments/assets/c5d42917-13b2-4747-bcec-9adc6492bc7b" />
<img width="1298" height="713" alt="Screenshot 2026-01-13 at 11 27 18" src="https://github.com/user-attachments/assets/8d68d7f9-6142-474f-b181-72040a35c83d" />

stevieflow commented:
@BibianaC thanks - looks good:

- I'm not sure what the large IATI logo at the top is for.  Is that intentional?
- Should there be a top right menu item for the Help docs?
- For top menu items, do we need to supply words for these (eg: something for dataDashboard.label ?)?
- I'm not sure what the drop down is meant to be with the "translations.selectCountry" label...
- The Useful links also need some attention, as they are misspelt etc

robredpath commented:
Thanks @BibianaC - this is good progress. I'd appreciate instructions on how to run this locally or how to access a dev instance, as there's only so much I can determine from a screenshot. In the meantime, are you able to provide screenshots of the other pages in the application? 

Specific things I can see so far:
* As @stevieflow mentions, Help Docs is missing from the header; these are at https://docs.countrydata.iatistandard.org/ 
* I assume the blank language switcher in the header is because this is a template?
* the "by Recipient Country or Region" etc links should be buttons like the current design, not a list of links. There are [a range of buttons](https://iati.github.io/design-system/?path=/docs/components-button--docs) in the design system that can be used here. 
* The "Access Data Files" section is misaligned. The design system has a [card gallery](https://iati.github.io/design-system/?path=/docs/components-card--docs) that might help, or we might want to make a version of the [data card](https://iati.github.io/design-system/?path=/docs/components-data-card--docs) that has a drop-down (cc @codemacabre )
* The section about guidance can be removed as that's all now in the Help Docs
* The "useful links" can be removed, and a link to GitHub and a note on licensing added, per other tools

BibianaC committed:
layouts: default: add Help Docs

BibianaC committed:
content: index: conver by ... links to buttons

BibianaC committed:
content: index: delete guidance section

BibianaC commented:
@stevieflow @robredpath @codemacabre 

I have added the latest screenshot to the description. I will always add the last one there. It is only one screenshot.

- I’m not sure what the large IATI logo at the top is for. Is that intentional? 
The big logo is not there; I don’t know why it appeared in the screenshots.
- Missing Help Docs: 
Added.
- For top menu items, do we need to supply words for these (eg: something for dataDashboard.label ?)? 
@robredpath ?
- I’m not sure what the drop down is meant to be with the "translations.selectCountry" label...
This is weird, I don't see any change in the code. @codemacabre do you know what happened?
- I assume the blank language switcher in the header is because this is a template? Also in the footer there is only English as an option.  
@codemacabre?
- The "by Recipient Country or Region" etc links should be buttons: 
Done.
- The "Access Data Files" section is misaligned. 
It isn’t; it is a different screenshot, sorry for the confusion.
- The section about guidance can be removed: 
Done.
- The "useful links" can be removed, and a link to GitHub and a note on licensing added, per other tools. 
@robredpath What do you want the note to say?

stevieflow commented:
Thanks @BibianaC - I see the screenshot at the top now.  All good - I think the remaining questions are with @robredpath & @codemacabre 

One thing I notice - in the screenshot the application specific stuff seems to be laid out differently to the live site:

<img width="1657" height="1102" alt="Country-Development-Finance-Data-01-15-2026_10_05_AM" src="https://github.com/user-attachments/assets/ae1586ae-4521-4a3b-bab2-76eb9c75bb03" />

That might just be a local issue, however

codemacabre commented:
@BibianaC I'm not sure what the issue is, possibly something to do with your setup, as it all appears to be working correctly on my end:

<img width="1737" height="1528" alt="image" src="https://github.com/user-attachments/assets/82bd91ae-890e-49f9-9965-1938624d2977" />

Note: I just re-cloned this from scratch and did the following:

1. `git clone git@github.com:iati-data-access/iati-data-access.github.io.git`
2. `cd iati-data-access.github.io`
3. `git checkout apply-design-system`
4. `npm i`
5. `npm run dev`

robredpath commented:
@BibianaC the screenshot @codemacabre shared is significantly different from the one at the top of the issue. Please let me know when this is stable so that I can review the latest work. 

Re: the contents of the footer, I'd follow https://dashboard.iatistandard.org as far as makes sense 

Is the rest of the application ready for review using the instruction that @codemacabre supplied?

IsabelBirds commented:
Hi, what does the about link lead to? That should be gone and totally replaced by the link to the help docs.
Also small spelling - it says "Satastore" in the footer.

BibianaC committed:
layouts: default: update footer

robredpath commented:
In the interest of getting this work wrapped up quicker, can we scope it to just apply to the header and footer, rather than the entire interface? The design system's [incremental migration](https://iati.github.io/design-system/?path=/docs/usage--docs#incremental-migration) functionality was designed for this, but that was intended to be used from the start of a project. 

I appreciate that changing implementation now might be harder than just pushing through and rolling this out more broadly, so I defer to @BibianaC and @codemacabre on whether this is actually helpful.

BibianaC commented:
@IsabelBirds the about link goes to: https://docs.countrydata.iatistandard.org/en/latest/ 
Do I delete it?

@robredpath this is the footer with the changes (using the Dashboard as a reference):

<img width="1479" height="416" alt="Screenshot 2026-01-16 at 11 22 48" src="https://github.com/user-attachments/assets/aad8707f-5dac-449d-963b-d74f99a56309" />

You can review using Myles instructions whenever you think it is ready for it.

robredpath commented:
@BibianaC thanks, but there's a few things that need to be updated in the footer:

* link to GitHub issues should actually link to GitHub issues
* link to the source code (per Dashboard footer) would still make sense
* License statement in the footer is GPLv3 but the repo states AGPL 

I also don't seem to have any text on the buttons:

<img width="1175" height="136" alt="Screenshot 2026-01-30 at 13 24 42" src="https://github.com/user-attachments/assets/da96c47f-25d7-4478-9616-43eb44bc7182" />

Also @IsabelBirds 's comment re: the About link still needs to be addressed. 

On the Dashboard and Custom Download pages there's a direct link to support@iatistandard.org that should be replaced with a link to the website's get-support page. I also note that there's a link to `/methodology` on those - presumably that content should be removed and replaced with a link to the help docs?

BibianaC committed:
layouts: default: fix url GitHub issues

BibianaC committed:
layouts: default: footer: add source code

BibianaC committed:
layouts: default: footer: fix license to correct one

BibianaC committed:
layouts: default: delete about

BibianaC commented:
@robredpath 


> link to GitHub issues should actually link to GitHub issues
> link to the source code (per Dashboard footer) would still make sense
> License statement in the footer is GPLv3 but the repo states AGPL

Done

> I also don't seem to have any text on the buttons:

@codemacabre do you know why it looks like this?

> Also @IsabelBirds 's comment re: the About link still needs to be addressed.

In my last comment, I double checked about this. I have now deleted the about link.

> On the Dashboard and Custom Download pages there's a direct link to [support@iatistandard.org](mailto:support@iatistandard.org) that should be replaced with a link to the website's get-support page. I also note that there's a link to /methodology on those - presumably that content should be removed and replaced with a link to the help docs?

I don't understand what you mean.

codemacabre commented:
> @codemacabre do you know why it looks like this?

@BibianaC There are a few issues here. Mostly that it's still got Bootstrap classes attached to the buttons, which are overriding the design system.

A quick and incomplete fix is to give the three button links the following classes: `btn btn-primary btn-block iati-button`. However, this makes them have a light cyan background on hover, which is inaccessible.

A better fix is to give all the button links the single class `iati-button`, but this means they're not the right width. You'd either need to make an addition to the design system for a new full-width button (i.e. which just needs the CSS `width: 100%;`) or add custom CSS / class to apply that width to those buttons.

BibianaC commented:
@codemacabre I think that Rob means that the text is not visible. He doesn't comment on the shape.

codemacabre commented:
@BibianaC I'm referring to the colour of the text (which is currently the same text and background colour), as well as the shape.

robredpath commented:
> On the Dashboard and Custom Download pages there's a direct link to [support@iatistandard.org](mailto:support@iatistandard.org) that should be replaced with a link to the website's get-support page. I also note that there's a link to /methodology on those - presumably that content should be removed and replaced with a link to the help docs?

> I don't understand what you mean.

On https://countrydata.iatistandard.org/data/recipient-country-or-region/AI/?filters=transaction_type%3A3,4,budget%3Byear%3A2025 :

<img width="1525" height="156" alt="Screenshot 2026-02-02 at 14 31 59" src="https://github.com/user-attachments/assets/443acbc1-af55-40c6-bd2b-361257d04619" />

The link to Methodology should link to the relevant section of the help docs and the email address should be replaced with a link to https://iatistandard.org/en/guidance/get-support/

BibianaC commented:
@robredpath thanks now I see what you mean.

Should the sentence 
> For support or training, reach out to [support@iatistandard.org](mailto:support@iatistandard.org)
 
be changed to 

> For support or training, reach out to [support](https://iatistandard.org/en/guidance/get-support/).

I have found where the wording of the part of the methodology is, but I can't find the link to change it. @codemacabre do you know where it is?

@codemacabre would you be able to fix the buttons?

robredpath commented:
> For support or training, reach out to [support](https://iatistandard.org/en/guidance/get-support/).

Actually, just remove that sentence. There's a Contact link in the header which suffices now.

BibianaC committed:
Delete 'For support or training, reach out to support@iatistandard.org'

BibianaC commented:
@robredpath contact link removed.

codemacabre committed:
content: index: replace b-btn with a; add custom iati-button-wide class

codemacabre commented:
@BibianaC I've replaced these four buttons with anchor links styled appropriately so you can read the text now:

<img width="1092" height="456" alt="image" src="https://github.com/user-attachments/assets/69ad8954-eb6c-4c89-8e72-8523e267be1b" />

BibianaC commented:
Thanks @codemacabre. Now what is left is the Methodology.

> The link to Methodology should link to the relevant section of the help docs and the email address should be replaced with a link to https://iatistandard.org/en/guidance/get-support/

codemacabre commented:
@BibianaC Oh I didn't realise you wanted me to do that bit too. I thought we'd got rid of the dropdown list in the About menu? Doesn't that mean the Methodology link has gone? And where is this email address that needs replacing?

BibianaC commented:
@codemacabre I can't find how to change it. 
The methodology is at the bottom of:
https://countrydata.iatistandard.org/data/recipient-country-or-region/AI/?filters=transaction_type%3A3,4,budget%3Byear%3A2025

codemacabre commented:
> The link to Methodology should link to the relevant section of the help docs

@robredpath @BibianaC Is the relevant section of the help docs just the [first item](https://docs.countrydata.iatistandard.org/en/latest/retrieval/) under the Methodology heading? There doesn't appear to be an actual Methodology page to link to.

stevieflow commented:
@codemacabre @BibianaC we will just link to the Help Docs from the establish menu item in the design system.  

_Within_ the help docs is a methodology, but we dont need to explicitly call this out in any other link or function.  People will just go to the Help doc in the same manner they go to Help docs in all other interfaces (top right hand button)

Thanks

codemacabre committed:
locales: update methodology statement to link to help docs

codemacabre commented:
As discussed with @stevieflow on a call, I've updated the methodology text to be "Additional details can be found in the help docs" (in all languages) with links to the main help docs.

This should now be ready for deployment.

BibianaC commented:
Thanks @codemacabre !!!

@robredpath I can't add you as a reviewer. If you think it is okay, you can merge the PR.

robredpath commented:
@BibianaC I don't have permission to merge this either.

codemacabre commented:
@BibianaC @robredpath Looks like I have permissions. Happy for me to merge?

robredpath commented:
@codemacabre yes please!
